x-pack/plugins/apm/server/lib/rum_client/ui_filters/get_es_filter.ts

```diff
@@ -23,5 +25,16 @@ export function getEsFilter(uiFilters: UIFilters) {
       };
     }) as ESFilter[];
 
+  if (
+    uiFilters.environment &&
+    uiFilters.environment !== ENVIRONMENT_ALL.value
+  ) {
+    mappedFilters.push({
+      term: {
+        [SERVICE_ENVIRONMENT]: uiFilters.environment,
+      },
+    });
+  }
+
   return mappedFilters;
```

Comment:
Perhaps use [`environmentQuery`](https://github.com/elastic/kibana/blob/e0a1a34fbc5a92cae76638514779a89cad853fa1/x-pack/plugins/apm/server/utils/queries.ts#L18-L30):

```suggestion
  return [...mappedFilters, ...environmentQuery(environment)]
```

Reply:
wow, thanks, that is much much better.